### PR TITLE
Adds the syndicate minigun.

### PR DESCRIPTION
**Pull request by Dorsisdwarf**
This PR adds the syndicate minigun, a heavy weapon (recoils if your other hand isn't free). It has a 400-round ammo drum which cannot be replaced, fires in bursts of 20 bullets, and does 7 damage a hit.

This gun is currently available by adminspawn and Summon Guns, but may be added to nuke ops.
minigun

Also adds myself to admins.txt, after like 5 months of coding.

### CODE REVIEW

**Comment by Dorsisdwarf** [comment]
What the fuck is commit "hell yeah" and why does it appear in all my branches.

File: code/game/gamemodes/wizard/rightandwrong.dm [hunk 1]
```diff
@@ -93,6 +94,15 @@
 					var/obj/item/weapon/gun/projectile/automatic/m90/gat
 					gat.pin = /obj/item/device/firing_pin
 					new gat(get_turf(H))
+				if("boltaction")
+					new /obj/item/weapon/gun/projectile/shotgun/boltaction(get_turf(H))
+				if("speargun")
+					new /obj/item/weapon/gun/projectile/automatic/speargun(get_turf(H))
+				if("minigun")
+					var/obj/item/weapon/gun/projectile/automatic/gatling/gat     // This looks like fun addition but it's secretly fun removal
```
Comment (Menshin) [inline code/game/gamemodes/wizard/rightandwrong.dm]: This have to be fixed (the runtime was present in the original file, see  #8197)

**Comment by WJohn** [comment]
Uh, 4 damage a bullet? I feel like you wouldn't really normally be standing after being shot by a hail of 20 bullets. Meanwhile we have a 50 shot LMG which does 60 damage per bullet. In terms of damage output, the LMG could deal 3000 damage per clip, whereas the minigun would do a way worse 1600 (that you can easily dodge) and can't be replaced. This thing needs to do WAY WAY more damage.

**Comment by Dorsisdwarf** [comment]
Lots of low-damage shots gets around armor really well - 2 clips will kill someone in full bulletproof vest.

I wouldn't want to take it above 9 - 10 damage will instakill an unarmored person.

**Comment by WJohn** [comment]
But you can easily just walk out of the way...

**Comment by Dorsisdwarf** [comment]
True  - but not if you're stunned (IE: The other four nuke ops) or in a crowd.

**Comment by Steelpoint** [comment]
I see someone played Payday 2...

All in all its neat, damage discussions/arguments aside.

**Comment by Incoming5643** [comment]
Nerfing a gun for summon guns is silly, that's where OP shit retires to when it becomes obvious it's imba.

**Comment by Dorsisdwarf** [comment]
A gun which can instantly kill the wizard is, perhaps, just a /little/ bit unfair for a single extra spell, especially when you factor in the 10 other guns being spawned

**Comment by Incoming5643** [comment]
Wand of death is already a threat a summon magic, so it's not like there's no premise for that sort of thing.

Between his hat and robe, a wizard reduces 20% of all bullet damage no matter where you're shooting, so a full power volley would do 64 damage, not 80.

Also if a wizard gets hit by EVERY bullet he kind of deserves it, it's not like he can't teleport at any moment most of the time.

**Comment by RemieRichards** [comment]
You removed the params argument from all the ```attackby```s you touched, intentionally or not, return them please.

**Comment by Lumipharon** [comment]
Don't volleys (or whatever the fuck we call them) all spawn from the same spot/go the same direction?
This isn't so much of an issue for 2 and 3 round volleys, but if it's true I can image this would be pretty retarded with a 20 round burst.

**Comment by Cheridan** [comment]
Suggestion: use the variance var on the minigun's magazine for that proper bulletspewer feel.

**Comment by Atskadan** [comment]
considering incoming intends to make a ton of super OP weapons for summon guns, i agree that the minigun should be better. even 6 damage would be good, that would do as much as a revolver if you took every single shot without armor.

**Comment by Incoming5643** [comment]
longtermproject

**Comment by Lumipharon** [comment]
just make it cause SLOW when you carry it, the ammo thing takes your backslot, and you need both hands to wield it.
Then make it do actual damage.

**Comment by WJohn** [comment]
It just seems weird to me that all other bullets, at their absolute worst, do at least 25 damage. Because, you know, it's a bullet.

You shouldn't be alive, at all, after being shredded by a minigun.

**Comment by Dorsisdwarf** [comment]
@cheridan , what is the 'variance' var?

On an unarmored target this gun already does 80 damage (60 in wiz) if you hit them with the whole burst.

Lumipharon, what are you talking about? Backpack ammo thing? I was told to drop that by miauw because the backpack items are snowflakey.

WJ, that's not true - bullet damage goes from 3 to 60 . 

I'll definitely up the damage to around 7 or so, which will crit anyone bad enough to be hit by the burst, while not outright killing them, and also probbably not killing people who've got bulletproof armor on.

**Comment by Lumipharon** [comment]
If it does that little damage, you'll be able to get augs and take zero damage with a bit of armor.
Even that aside if it only does 80 damage from that many hits, then it's bad. Why would I use this bulky two hander when I could use a revolver and 2 hit crit people?

File: code/game/gamemodes/wizard/rightandwrong.dm [hunk 2]
```diff
@@ -66,33 +66,41 @@
...
+					var/obj/item/weapon/gun/energy/kinetic_accelerator/crossbow/large/gat = new(get_turf(H))
 					gat.pin = /obj/item/device/firing_pin
-					new gat(get_turf(H))
+
 				if("nuclear")
-					var/obj/item/weapon/gun/energy/gun/nuclear/gat
-					gat.pin = /obj/item/device/firing_pin
-					new gat(get_turf(H))
+					var/obj/item/weapon/gun/energy/gun/nuclear/gat = new(get_turf(H))
+					gat.pin = /obj/item/device/firing_pin //no authentication pins for spawned guns. fun allowed. PRAISE THE CODERBUS
```
Comment (Menshin) [inline code/game/gamemodes/wizard/rightandwrong.dm]: You forgot to change
```gat.pin = /obj/item/device/firing_pin```
to
```gat.pin = new /obj/item/device/firing_pin```

**Comment by Cheridan** [comment]
variance is used to give 'spread' to bullets, it's currently used in buckshot shells.

**Comment by Dorsisdwarf** [comment]
@Lumipharon That's not how armor works. Armor values reduce a % off each bit of damage taken, rounding down to avoid taking 0.5 damage.

So a bulletproof vest with 75 bulletproof turns a 100-damage bullet into a 75-damage one, but turns a 10-damage bullet into a 7 one.

Giving this spread is a good idea, yeah, it's a crowd control weapon.

**Comment by Lumipharon** [comment]
Dorf, augments are differen from armor, and you stack them together to get mlgpr0 damage immunity.

**Comment by Dorsisdwarf** [comment]
Augments are Brute-protect, not bullet protect anyway, aren't they? 10 built-in brute armor for the full set.

**Comment by Lumipharon** [comment]
While I could very well be wrong, looking at the code for augs it simply refers to burn and brute damage, so I'd think that doesn't make the distinctions that armor does.

**Comment by RemieRichards** [comment]
Augments remove 5 brute and 4(might be 5) Burn damage per attack.
Every time.

So 10 Brute --> 5 Brute
7 Brute --> 2 Brute etc.

**Comment by Dorsisdwarf** [comment]
I was under the impression that augmentations actually functioned like armor. They certainly used to, because that was how the 100% brute resist bug worked (fully upgraded golaith + secsuit + full augs), but I guess that was changed.

Even so, during nuke and wizard, I doubt the 1-2 people it's physically possible to augment (assuming someone wants an augment, assuming robotics knows how to augment, assuming robotics want to spend a tonne of metal of some greyshirt) are unlikely to have a serious impact on the usage of this crowd control weapon.

**Comment by RemieRichards** [comment]
@Dorsisdwarf 
>I was under the impression that augmentations actually functioned like armor. They certainly used to

>certainly used to

No this is false, they never did, they've always taken a flat 5/4 damage off of every attack.

the 100% resist bug comes from the fact that you're left with about 5 damage left in an attack with the max armour values you can get, with augs taking a flat 5 every time, it made it 0.

**Comment by Dorsisdwarf** [comment]
Huh. I'll put this up to 10 damage when I get home.

**Comment by Dorsisdwarf** [comment]
Well looks like this is the last commit for this  PR because git has actually set fire to my fucking commits when I tried to resolve the fucking conflicts

**Comment by Dorsisdwarf** [comment]
RIP. This will teach me to work faster in future, so that I don't end up with a zillion conflicts

**Pull request closed without merging**